pageserver/src/object_repository.rs

```diff
@@ -159,6 +158,11 @@ impl Repository for ObjectRepository {
     }
 }
 
+struct RelMetadata {
+    size: Option<u32>,
```

Comment:
Is `size` ever None

Reply:
Bump on this

Reply:
Sorry, it seems to me that I have answered, but looks like my message was not sent.
Empty size in RelMetadata indicates dropped relation. The main branch doesn't support dropping of relations and this is why this option is not used now. But once it will be supported relation cache can handle it (it is already checked in get_rel_exists)

Reply:
Please put that in a comment here

Reply:
Done. After more thinking I decided that Patrick was right: there is no sense to cache dropped relation metadata. We in any case need to remove cached relation metadata for dropped relation to avoid memory overflow. And as far as we are caching only metadata of most recent version of relation, it is reasonable to delete metadata immediately when relation is dropped.